src/Feature/Layout/Model.re

```diff
@@ -132,57 +132,80 @@ type model = {
   activeGroupId: int,
 };
 
+let activeTree = layout =>
+  switch (layout.uncommittedTree) {
+  | `Resizing(tree)
+  | `Maximized(tree) => tree
+  | `None => layout.tree
+  };
+
+type model = {
+  layouts: list(layout),
+  activeLayoutIndex: int,
+};
+
 let initial = editors => {
   let initialGroup = Group.create(editors);
 
-  {
+  let initialLayout = {
     tree: Layout.singleton(initialGroup.id),
     uncommittedTree: `None,
     groups: [initialGroup],
     activeGroupId: initialGroup.id,
   };
+
+  {layouts: [initialLayout], activeLayoutIndex: 0};
 };
 
-let groupById = (id, model) =>
-  List.find_opt((group: Group.t) => group.id == id, model.groups);
+let groupById = (id, layout) =>
+  List.find_opt((group: Group.t) => group.id == id, layout.groups);
+
+let activeLayout = model => List.nth(model.layouts, model.activeLayoutIndex);
 
 let activeGroup = model =>
-  groupById(model.activeGroupId, model) |> Option.get;
+  model
+  |> activeLayout
+  |> (layout => groupById(layout.activeGroupId, layout) |> Option.get);
 
 let activeEditor = model => model |> activeGroup |> Group.selected;
 
-let activeTree = model =>
-  switch (model.uncommittedTree) {
-  | `Resizing(tree)
-  | `Maximized(tree) => tree
-  | `None => model.tree
-  };
-
-let updateTree = (f, model) => {
+let updateActiveLayout = (f, model) => {
   ...model,
-  tree: f(activeTree(model)),
-  uncommittedTree: `None,
-};
-
-let updateActiveGroup = (f, model) => {
-  ...model,
-  groups:
-    List.map(
-      (group: Group.t) => group.id == model.activeGroupId ? f(group) : group,
-      model.groups,
+  layouts:
+    List.mapi(
+      (i, layout) => i == model.activeLayoutIndex ? f(layout) : layout,
+      model.layouts,
     ),
 };
 
-let windows = model => Layout.windows(activeTree(model));
+let updateTree = f =>
+  updateActiveLayout(layout =>
+    {...layout, tree: f(activeTree(layout)), uncommittedTree: `None}
+  );
+
+let updateActiveGroup = f =>
+  updateActiveLayout(layout =>
+    {
+      ...layout,
+      groups:
+        List.map(
+          (group: Group.t) =>
+            group.id == layout.activeGroupId ? f(group) : group,
+          layout.groups,
+        ),
+    }
+  );
+
+let windows = model => Layout.windows(model |> activeLayout |> activeTree);
 
 let visibleEditors = model =>
   model
   |> windows
-  |> List.filter_map(id => groupById(id, model))
+  |> List.filter_map(id => model |> activeLayout |> groupById(id))
```

Comment:
Nice! Seems like adding the layout tabs was a pretty straightforward addition to the model, with the groundwork you laid out in #1977